### PR TITLE
PEP8とFlake8に基づいて修正

### PR DESCRIPTION
**Pull request by sfc-quantan**

### CODE REVIEW

File: create_testdata.py [hunk 1]
```diff
@@ -3,38 +3,33 @@
...
-k=1367
-width=480
-height=640
-gominum=7
-xywhc=5
+k = 1367
+width = 480
+height = 640
+gominum = 7
+xywhc = 5
```
Comment (makora9143) [inline create_testdata.py]: 定数は変数名を全部大文字にしたほうがいいと思います

File: create_testdata.py [hunk 2]
```diff
@@ -3,38 +3,33 @@
...
-    f=open("./1/annotations/"+str(i)+".json","r")
-    json_dict=json.load(f)
-    tmp = len(json_dict["objects"])    
+for i in range(1, k+1):
+    f = open("./1/annotations/"+str(i)+".json", "r")
+    json_dict = json.load(f)
+    tmp = len(json_dict["objects"])
     for j in range(tmp):
-        y[i-1][j]=json_dict["objects"][j]["x_y_w_h"]+[1]
+        y[i-1][j] = json_dict["objects"][j]["x_y_w_h"]+[1]
```
Comment (makora9143) [inline create_testdata.py]: for文の入れ子は計算量が$O(n^2)$になるので，numpyの関数を使ったりして，
できるだけ工夫してください．

File: evaluate_model.py [hunk 3]
```diff
@@ -1,43 +1,42 @@
-from keras.models import Model,load_model
+# coding: utf-8
+from keras.models import load_model
 import json
 import random
 import numpy as np
 from PIL import Image
 from loss import org_mse
```
Comment (makora9143) [inline evaluate_model.py]: 一般にモジュールの読み込みは，
1．Pythonの組み込みモジュール
2．サードパーティモジュール
3．自作モジュール
なので，その順番にしましょう

File: import_data.py [hunk 4]
```diff
@@ -1,62 +1,59 @@
...
-
-    y_c_train=y_c_train.reshape(-1,7)
-    y_c_test=y_c_test.reshape(-1,7)
-    
-    del x,y 
+            y_xywh_test[i][j][0] = y_xywh_test[i][j][0]/640
+            y_xywh_test[i][j][1] = y_xywh_test[i][j][1]/480
+            y_xywh_test[i][j][2] = y_xywh_test[i][j][2]/640
+            y_xywh_test[i][j][3] = y_xywh_test[i][j][3]/480
+
```
Comment (makora9143) [inline import_data.py]: 34-39行目と同じコードなので，
ここは関数化したほうがいいと思います

File: import_data.py [hunk 5]
```diff
@@ -1,62 +1,59 @@
...
+    x_train = x_train.astype('float32')/255
+    x_test = x_test.astype('float32')/255
+    y_xywh_train = y_xywh_train.reshape(-1, 28).astype('float32')
+    y_xywh_test = y_xywh_test.reshape(-1, 28).astype('float32')
+
+    y_c_train = y_c_train.reshape(-1, 7)
+    y_c_test = y_c_test.reshape(-1, 7)
+
+    del x, y
     gc.collect()
```
Comment (makora9143) [inline import_data.py]: この2行を入れた理由はあります？

File: loss.py [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
+    y_true = K.reshape(y_true, (-1, 7, 4))
+    xy = y_true[:, :, 0:2]
+    wh = y_true[:, :, 2:4]
 
-   temp=K.square(xy2-xy)
-   temp=temp+K.square(K.sqrt(wh2)-K.sqrt(wh))
-   return temp
+    y_pred = K.reshape(y_pred, (-1, 7, 4))
+    xy2 = y_pred[:, :, 0:2]
+    wh2 = y_true[:, :, 2:4]
```
Comment (makora9143) [inline loss.py]: xy2とwh2の”2”はどういう意味？？

File: loss.py [hunk 7]
```diff
@@ -1,16 +1,16 @@
...
 
+    temp = K.square(xy2-xy)
+    temp = temp + K.square(K.sqrt(wh2)-K.sqrt(wh))
```
Comment (makora9143) [inline loss.py]: 直しきれてない

File: main.py [hunk 8]
```diff
@@ -1,48 +1,54 @@
...
-from keras import metrics
 from keras.optimizers import Adam
 from keras.models import Model
-from keras.callbacks import ModelCheckpoint,EarlyStopping
-from keras import backend as K
+from keras.callbacks import ModelCheckpoint, EarlyStopping
 from loss import org_mse
 from import_data import make_data
 from model import make_model
 from evaluate_model import evaluate1
```
Comment (makora9143) [inline main.py]: モジュールの読み込み順

File: main.py [hunk 9]
```diff
@@ -1,48 +1,54 @@
...
 
 
-batch_size=8
-epoch=1
+batch_size = 8
+epoch = 1
```
Comment (makora9143) [inline main.py]: 定数

File: main.py [hunk 10]
```diff
@@ -1,48 +1,54 @@
...
-	loss_weights={'main_output': 1., 'side_output': 0.2,'numbers_output':1})
+early_stopping = EarlyStopping(
+    monitor='val_loss', min_delta=0, patience=190, verbose=0, mode='auto')
 
-model.fit(x=x_train,y=[y_xywh_train,y_c_train,z_train],batch_size=batch_size,epochs=epoch,verbose=1,
-        validation_data=(x_test,[y_xywh_test,y_c_test,z_test]),callbacks=[checkpointer,early_stopping])
+model.compile(
+    optimizer=adam,
+    loss={'main_output': org_mse, 'side_output': 'binary_crossentropy',
+          'numbers_output': 'categorical_crossentropy'},
```
Comment (makora9143) [inline main.py]: 好みだけど，

```python
loss = {
    'main_output': org_mse,
    'side_output': 'binary_crossentropy',
    'numbers_output': 'categorical_crossentropy'
},
```

か，

```python
loss = {'main_output': org_mse,
             'side_output': 'binary_crossentropy',
             'numbers_output': 'categorical_crossentropy'},
```

ってしたほうが良い？

File: main.py [hunk 11]
```diff
@@ -1,48 +1,54 @@
...
+    metrics={'main_output': 'mae', 'side_output': 'accuracy',
+             'numbers_output': 'accuracy'},
+    loss_weights={'main_output': 1., 'side_output': 0.2, 'numbers_output': 1})
```
Comment (makora9143) [inline main.py]: 上と同様

File: main.py [hunk 12]
```diff
@@ -1,48 +1,54 @@
...
+          batch_size=batch_size, epochs=epoch, verbose=1,
+          validation_data=(x_test, [y_xywh_test, y_c_test, z_test]),
+          callbacks=[checkpointer, early_stopping])
+
+score = model.evaluate(
+    x_test, [y_xywh_test, y_c_test, z_test], verbose=1, batch_size=8)
+print(score[0], score[1])
 
 json_string = model.to_json()
 open('model.json', 'w').write(json_string)
```
Comment (makora9143) [inline main.py]: この書き方だと，ファイルオブジェクトがopenしっぱなしなので，
writeの後にcloseする必要があります．
以下の書き方が便利です．

```python
with open('model.json', 'w') as f:
    f.write(json_string)
```

としたほうがいいです．

File: evaluate_model.py [hunk 13]
```diff
@@ -1,43 +1,42 @@
...
-	    test1[i][2]=test1[i][2]*640
-	    test1[i][3]=test1[i][3]*480
-
-
-	print(test1.astype('int32'))
-	print(test2)
-	for i in range(len(json_dict["objects"])):
-	    print(json_dict["objects"][i]["x_y_w_h"])
 
+def evaluate1():
```
Comment (makora9143) [inline evaluate_model.py]: `evaluate1()`の1ってなんでしょう

File: main.py [hunk 14]
```diff
@@ -1,48 +1,54 @@
...
 from import_data import make_data
 from model import make_model
 from evaluate_model import evaluate1
 
 
-batch_size=8
-epoch=1
+batch_size = 8
+epoch = 1
 gpu_config.set_tensorflow([1])
```
Comment (makora9143) [inline main.py]: これはkerasモジュールを読み込む前に入れたほうがいいです．

File: create_testdata.py [hunk 15]
```diff
@@ -3,38 +3,33 @@
...
-k=1367
-width=480
-height=640
-gominum=7
-xywhc=5
+k = 1367
+width = 480
+height = 640
+gominum = 7
+xywhc = 5
```
Comment (makora9143) [inline create_testdata.py:10]: 定数は変数名を全部大文字にしたほうがいいと思います

File: create_testdata.py [hunk 16]
```diff
@@ -3,38 +3,33 @@
...
-    f=open("./1/annotations/"+str(i)+".json","r")
-    json_dict=json.load(f)
-    tmp = len(json_dict["objects"])    
+for i in range(1, k+1):
+    f = open("./1/annotations/"+str(i)+".json", "r")
+    json_dict = json.load(f)
+    tmp = len(json_dict["objects"])
     for j in range(tmp):
-        y[i-1][j]=json_dict["objects"][j]["x_y_w_h"]+[1]
+        y[i-1][j] = json_dict["objects"][j]["x_y_w_h"]+[1]
```
Comment (makora9143) [inline create_testdata.py:25]: for文の入れ子は計算量が$O(n^2)$になるので，numpyの関数を使ったりして，
できるだけ工夫してください．

File: evaluate_model.py (hunk 3 above)
Comment (makora9143) [inline evaluate_model.py:7]: 一般にモジュールの読み込みは，
1．Pythonの組み込みモジュール
2．サードパーティモジュール
3．自作モジュール
なので，その順番にしましょう

File: evaluate_model.py (hunk 13 above)
Comment (makora9143) [inline evaluate_model.py:10]: `evaluate1()`の1ってなんでしょう

File: import_data.py (hunk 4 above)
Comment (makora9143) [inline import_data.py:47]: 34-39行目と同じコードなので，
ここは関数化したほうがいいと思います

File: import_data.py (hunk 5 above)
Comment (makora9143) [inline import_data.py:57]: この2行を入れた理由はあります？

File: loss.py (hunk 6 above)
Comment (makora9143) [inline loss.py:12]: xy2とwh2の”2”はどういう意味？？

File: loss.py (hunk 7 above)
Comment (makora9143) [inline loss.py:15]: 直しきれてない

File: main.py (hunk 8 above)
Comment (makora9143) [inline main.py:9]: モジュールの読み込み順
ただしimport gpu_configだけは例外で

File: main.py (hunk 9 above)
Comment (makora9143) [inline main.py:13]: 定数

File: main.py (hunk 14 above)
Comment (makora9143) [inline main.py:14]: これはkerasモジュールを読み込む前に入れたほうがいいです．

File: main.py (hunk 10 above)
Comment (makora9143) [inline main.py:35]: 好みだけど，

```python
loss = {
    'main_output': org_mse,
    'side_output': 'binary_crossentropy',
    'numbers_output': 'categorical_crossentropy'
},
```

か，

```python
loss = {'main_output': org_mse,
             'side_output': 'binary_crossentropy',
             'numbers_output': 'categorical_crossentropy'},
```

ってしたほうが良い？

File: main.py (hunk 11 above)
Comment (makora9143) [inline main.py:38]: 上と同様

File: main.py (hunk 12 above)
Comment (makora9143) [inline main.py:50]: この書き方だと，ファイルオブジェクトがopenしっぱなしなので，
writeの後にcloseする必要があります．
以下の書き方が便利です．

```python
with open('model.json', 'w') as f:
    f.write(json_string)
```

としたほうがいいです．

File: main.py [hunk 17]
```diff
@@ -1,48 +1,54 @@
...
+    x_test, [y_xywh_test, y_c_test, z_test], verbose=1, batch_size=8)
+print(score[0], score[1])
 
 json_string = model.to_json()
 open('model.json', 'w').write(json_string)
 model.save_weights('param.hdf5')
-model.save('model.h5') 
+model.save('model.h5')
 
 evaluate1()
```
Comment (makora9143) [inline main.py:54]: そもそも直書きはそんなに好ましくないので，

```python
def main():
    (x_train, x_test, y_xywh_train, y_xywh_test, y_c_train, y_c_test, z_train, z_test) = make_data()
    ...

if __name__ == "__main__":
    main()
    
```

とするのが良いです．

**Review by makora9143 — COMMENTED** [review]
だいたい良くなってきているけど，もう少し効率の良い書き方にできるといいですね．

File: create_testdata.py (hunk 15 above)
Comment (sfc-quantan) [reply create_testdata.py:10]: ○

File: evaluate_model.py (hunk 3 above)
Comment (sfc-quantan) [reply evaluate_model.py:7]: ○

File: evaluate_model.py (hunk 13 above)
Comment (sfc-quantan) [reply evaluate_model.py:10]: evaluateだと予約言語があるのではと思って1を入れてしまいました。修正しました。

File: import_data.py (hunk 5 above)
Comment (sfc-quantan) [reply import_data.py:57]: いや、keras.utils.to_categorical(z, 8)でちゃんとnx８の行列になっているので必要なかったです。それに7ではなくて８ですね。間違っていました。ありがとうございます。

File: loss.py (hunk 6 above)
Comment (sfc-quantan) [reply loss.py:12]: xyが実測値でxy2の方を予想値として区別するために2をつけたのですがわかりにくいですね。
xyをxy_true、xy2をxy_predと修正しました。

File: loss.py (hunk 7 above)
Comment (sfc-quantan) [reply loss.py:15]: ○

File: main.py (hunk 8 above)
Comment (sfc-quantan) [reply main.py:9]: ○

File: main.py (hunk 9 above)
Comment (sfc-quantan) [reply main.py:13]: ○

File: main.py (hunk 14 above)
Comment (sfc-quantan) [reply main.py:14]: ○

File: main.py (hunk 10 above)
Comment (sfc-quantan) [reply main.py:35]: ○

File: main.py (hunk 11 above)
Comment (sfc-quantan) [reply main.py:38]: ○

File: main.py (hunk 12 above)
Comment (sfc-quantan) [reply main.py:50]: ○

File: main.py (hunk 17 above)
Comment (sfc-quantan) [reply main.py:54]: ○

File: import_data.py (hunk 4 above)
Comment (sfc-quantan) [reply import_data.py:47]: ○
